### PR TITLE
Remove pygresql from gp7

### PR DESCRIPTION
**Pull request by zhrt123**
This patch removes code related to pygresql from gp7 since pygresql is replaced by psycopg2 in #15988.

## Notice
We should wait for #15988 to be merged.
TODO:
- fix the comment related to pygresql.
- remove pyyaml and psutil from pythonSrc and install them by yum.
- remove all the code related with with_pythonsrc_ext

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: README.Rhel-Rocky.bash [hunk 1]
```diff
@@ -8,11 +8,11 @@ for i in $INSTALL_PKGS; do
...
 done
 
-# Needed for pygresql, or you can source greenplum_path.sh after compiling database and installing python-dependencies then
+# Needed for psycopg2, or you can source greenplum_path.sh after compiling database and installing python-dependencies then
 sudo yum install -y postgresql 
 sudo yum install -y postgresql-devel
+sudo yum install -y python3-psycopg2
 
-pip3.9 install pygresql
 pip3.9 install -r python-dependencies.txt
```
Comment (beeender) [inline README.Rhel-Rocky.bash:16]: `pip3.9 install -r python-dependencies.txt` this can be removed, as well as the file. And add `python3-psutil`/`python3-pyyaml`.

File: README.Rhel-Rocky.bash [hunk 2]
```diff
@@ -8,11 +8,11 @@ for i in $INSTALL_PKGS; do
   sudo yum install -y $i
 done
 
-# Needed for pygresql, or you can source greenplum_path.sh after compiling database and installing python-dependencies then
+# Needed for psycopg2, or you can source greenplum_path.sh after compiling database and installing python-dependencies then
```
Comment (beeender) [inline README.Rhel-Rocky.bash:11]: I think line 11, 12 and 13 can be just removed. We don't need them directly.

File: gpAux/Makefile [hunk 3]
```diff
@@ -368,11 +367,7 @@ define tmpCLIENTS_FILESET_BIN
 endef
 CLIENTS_FILESET_BIN = $(strip $(tmpCLIENTS_FILESET_BIN))
 
-# pg.py, pgdb.py, _pg.so are from pygresql which does not install into a single module
 define tmpCLIENTS_FILESET_BINEXT
-	pg.py
-	pgdb.py
-	_pg*.so
 	yaml
```
Comment (beeender) [inline gpAux/Makefile:371]: maybe yaml can be removed as well. Let's test this for windows.

File: gpMgmt/bin/Makefile [hunk 4]
```diff
@@ -59,7 +59,7 @@ core:
 	python3 gpconfig_modules/parse_guc_metadata.py $(DESTDIR)$(prefix)
 
 ifeq ($(with_pythonsrc_ext), yes)
```
Comment (beeender) [inline gpMgmt/bin/Makefile:61]: I think we can just remove all the code related with `with_pythonsrc_ext`?

File: README.Rhel-Rocky.bash (hunk 2 above)
Comment (zhrt123) [reply README.Rhel-Rocky.bash:11]: Yes, we will remove these comments after the CI run green.

File: gpMgmt/bin/Makefile (hunk 4 above)
Comment (zhrt123) [reply gpMgmt/bin/Makefile:61]: I think so, since this dir will be clean.

File: README.Rhel-Rocky.bash (hunk 1 above)
Comment (zhrt123) [reply README.Rhel-Rocky.bash:15]: I think we can do this in another PR. Since this PR just remove pygresql from gp7.

Comment (beeender) [reply README.Rhel-Rocky.bash:15]: OK, don't forget remove below in the other PR

```
sudo yum install -y postgresql 
sudo yum install -y postgresql-devel
```

Those are not needed since they are just the build dependencies for `pygresql`.

File: README.Ubuntu.bash [hunk 5]
```diff
@@ -36,7 +36,6 @@ sudo apt-get install -y \
 	python3-dev \
 	python3-pip \
 	python3-psutil \
```
Comment (beeender) [inline README.Ubuntu.bash:38]: Add `python3-psycopg2`

File: README.Windows.md [hunk 6]
```diff
@@ -142,12 +142,9 @@ cmake --build . --config Release --target ALL_BUILD
 cmake --build . --config Release --target INSTALL
 ```
 
-4. Build pygresql, needed by gpload
+4. Build psycopg2, needed by gpload
```
Comment (beeender) [inline README.Windows.md:145]: it is not `Build` anymore. Rephrase it to:

```
4. Install psycopg2 and pyyaml, needed by gpload
```

File: concourse/scripts/combine_cli_coverage.bash [hunk 7]
```diff
@@ -83,7 +83,6 @@ omit =
     */site-packages/*
     */bin/behave
     */python/psutil/*
-    */python/pygresql/*
     */python/subprocess32.py
     */python/yaml/*
```
Comment (beeender) [inline concourse/scripts/combine_cli_coverage.bash:87]: `psutil`, `subprocess32` and `yaml` can be removed as well I think? I guess we are going to handle it in another PR?

File: gpAux/Makefile (hunk 3 above)
Comment (beeender) [reply gpAux/Makefile:371]: yaml can be removed, we did the test

File: gpAux/client/install/src/windows/greenplum-clients.wxs [hunk 8]
```diff
@@ -1140,12 +1140,6 @@ If you want to review or change any of your installation settings, click Back. C
             </Directory>
             <Directory Id="TOPLEVELlib" Name="lib">
               <Directory Id="PYTHONlib" Name="python">
-                <Component Id="PYGRESQLlib" DiskId="1" Guid="CDFE5977-8F11-43AF-95B7-374510FA872D">
-                  <File Id="__init__.py" Name="__init__.py" Source="$(var.SRCDIR)\lib\python\__init__.py" />
-                  <File Id="_pg.pyd" Name="_pg.pyd" Source="$(var.SRCDIR)\lib\python\_pg.pyd" />
-                  <File Id="pg.py" Name="pg.py" Source="$(var.SRCDIR)\lib\python\pg.py" />
-                  <File Id="pgdb.py" Name="pgdb.py" Source="$(var.SRCDIR)\lib\python\pgdb.py" />
-                </Component>
                 <Directory Id="YAMLlib" Name="yaml">
```
Comment (beeender) [inline gpAux/client/install/src/windows/greenplum-clients.wxs:1143]: remove yaml?

File: gpAux/client/install/src/windows/greenplum-clients.wxs [hunk 9]
```diff
@@ -1179,7 +1173,6 @@ If you want to review or change any of your installation settings, click Back. C
       <Feature Id="psql" Level="1" Title="Greenplum DB client command line tools" Description="Greenplum DB client command line tools" AllowAdvertise="no" Display="expand">
         <ComponentRef Id="TOPLEVELbin" />
         <ComponentRef Id="GPPYlib" />
-        <ComponentRef Id="PYGRESQLlib" />
         <ComponentRef Id="YAMLlib" />
```
Comment (beeender) [inline gpAux/client/install/src/windows/greenplum-clients.wxs:1176]: ditto

File: gpdb-doc/markdown/admin_guide/access_db/topics/g-database-application-interfaces.html.md [hunk 10]
```diff
@@ -12,6 +12,7 @@ You may want to develop your own client applications that interface to Greenplum
 |JDBC|pgjdbc|[https://jdbc.postgresql.org/](https://jdbc.postgresql.org/)|
 |Perl DBI|pgperl|[https://metacpan.org/release/DBD-Pg](https://metacpan.org/release/DBD-Pg)|
 |Python DBI|pygresql|[http://www.pygresql.org/](http://www.pygresql.org/)|
```
Comment (beeender) [inline gpdb-doc/markdown/admin_guide/access_db/topics/g-database-application-interfaces.html.md:14]: remove pygresql?

**Comment by zhrt123** [comment]
This PR is closed since #16091 is merged.
